caffe2/operators/asin_op.cc

```diff
@@ -0,0 +1,71 @@
+#include "caffe2/operators/elementwise_op.h"
+#include "caffe2/utils/math.h"
+
+namespace caffe2 {
+
+struct AsinCPUFunctor {
+  template <typename T>
+  inline void
+  operator()(const int n, const T* x, T* y, CPUContext* device_context) {
+    math::Asin<T, CPUContext>(n, x, y, device_context);
+  }
+};
+
+REGISTER_CPU_OPERATOR(
+    Asin,
+    UnaryElementwiseOp<TensorTypes<float>, CPUContext, AsinCPUFunctor>);
+
+OPERATOR_SCHEMA(Asin)
+    .NumInputs(1)
+    .NumOutputs(1)
+    .IdenticalTypeAndShape()
+    .SetDoc(R"DOC(
+Calculates the arcsine of the given input tensor, element-wise.
+)DOC")
+    .Input(0, "input", "Input tensor")
+    .Output(0, "output", "The arcsine of the input tensor computed element-wise");
+
+OPERATOR_SCHEMA(AsinGradient).NumInputs(2).NumOutputs(1).IdenticalTypeAndShape();
+
+class GetAsinGradient : public GradientMakerBase {
+  using GradientMakerBase::GradientMakerBase;
+  vector<OperatorDef> GetGradientDefs() override {
+    Argument one_arg;
+    one_arg.set_name("value");
+    one_arg.set_f(1);
+    return vector<OperatorDef>{CreateOperatorDef(
```

Comment:
how about just fusing this manually? this implementation on Caffe2 graph seems to be  inefficient. And we are not there for full compilers support yet.